File: P1.3/P1.3_opcional/IVAFilter.dart
import 'Target.dart';
import 'Filter.dart';

class IVAFilter implements Filter{

  execute(Target t){

    double iva = aletorio(3,30);

    String a = 'Aplicando filtro de IVA ' + toString(iva) + '%';

    t.addLog(a);
    t.setPrecio(t.precio + t.precio * iva/100);

  }

}

File: P1.3/P1.3_obligatorio/src/IVAFilter.cpp
#include "color.h"
#include "IVAFilter.h"
#include <string>
using namespace std;

void IVAFilter::execute(Target& t) {
    double iva = Filter::aleatorio(3, 30);
    string a = "Aplicando filtro de IVA " + to_string(iva) + "%";
    t.addLog(a);
    t.setPrecio( t.getPrecio() + t.getPrecio() * iva/100 );
}